src/api/usersRouter.ts

```diff
@@ -62,15 +62,36 @@ usersRouter.post('/:userId/resendInvitation', async (req: UserRequest, res) => {
       return res.sendStatus(404);
     }
     if (resultLength > 1) {
-        errorLogger.error(
-          `Multiple results received when loading user entry for ${req.user?.email}`
-        );
+      errorLogger.error(`Multiple results received when loading user entry for ${req.user?.email}`);
       return res.sendStatus(500);
     }
 
     logger.info(`Resending invitation email for ${req.user?.email}, keycloak ID ${user[0].id}`);
     await sendInviteEmail(kcAdminClient, user[0]);
     return res.sendStatus(200);
-});
+  });
+
+  usersRouter.delete('/:userId', async (req: UserRequest, res) => {
+    const { user } = req;
+    if (req.auth?.payload?.email === user?.email) {
+      return res.status(403).send([{ message: 'You do not have permission to delete yourself.' }]);
+    }
+    await user!.$query().delete();
```

Comment:
I just realised this will still leave the user with a keycloak account - so they could potentially still get a valid JWT and possibly even make API calls. This should delete the user in keycloak as well.